edu.wpi.first.wpilib.plugins.cpp/src/main/java/edu/wpi/first/wpilib/plugins/cpp/WPILibCPPPlugin.java

```diff
@@ -81,8 +88,36 @@ public void earlyStartup() {
 		new CPPInstaller(getCurrentVersion()).installIfNecessary(true);
 		Properties props = WPILibCore.getDefault().getProjectProperties(null);
     	WPILibCore.getDefault().saveGlobalProperties(props);
+		updateVariables();
 	}
 
+	@Override
+	public void startup() {
+		
+	}
+	
+	private void updateVariables() {
+		try {
+			
+			IUserVarSupplier varSupplier = CCorePlugin.getDefault().getUserVarSupplier();
+			IStorableCdtVariables vars = varSupplier.getWorkspaceVariablesCopy();
+			
+			File dir = new File(WPILibCore.getDefault().getWPILibBaseDir() + File.separator + "user" + File.separator + "cpp" + File.separator + "lib");
+			File[] filesList = dir.listFiles();
+			String libNames = "";
+			for (File file : filesList) {
+				if (file.isFile()) {
+					libNames += "-l" + file.getName().substring(3).split("[.]")[0] + " ";
```

Comment:
1. I know the convention is, on Linux, libs start with lib, but do we want to consider adding a check to make sure that the file correctly does this?

2. We should split on the last occurrence of `.`, not the first.